### ISSUE TITLE
Clicking current board link in index doesn't soft refresh the index (+ catalog clusterfuck)

### ISSUE THREAD

**Issue by Daiz** [ccd0/4chan-x#306]
~~I use 4chan X's catalog mode in the index. Navigating to a different board changes the index mode to Paged. 4chan X should really remember your selected index mode so I wouldn't have to change it every time. This used to work earlier but seems to have broken at some point.~~

Another thing that used to work was that clicking the current board link when at the index would also refresh the index, which is something I used all the time. ~~Now it just switches the index mode to paged and doesn't even refresh the index.~~ **Now depending on your settings it might do nothing, change the index mode or just reload the whole page.**

Okay, further looking into the options I figured out how this works. So in 4chan X options, you have [1.1] JSON Navigation -> Use 4chan X Catalog and [1.2] Catalog Links. Then, if you have [1.2] enabled, you get a setting in the header menu for [2] Catalog links.

Now, if you have [1.1] enabled and [1.2] disabled, then things work like they used to in Mayhem's 4chan X. Keeping the regular index in 4chan X catalog mode means your stuff remains in catalog mode, and all is nice. However, if you enable [1.2], things get messy. Now you have [2], and if you have it off, then you have a "saved" index mode out of the three "regular options" that is remembered and that is switched to if you're in catalog mode and press a board link. If you enable [2] and go to 4chan X catalog mode, the board link does nothing, unless you're in one of the regular index modes, in which case it switches to catalog mode. New boards load in 4chan X catalog mode by default. Of course, if [1.1] is disabled, then you'd get regular 4chan catalog instead.

This whole thing is, to put it bluntly, a pretty ridiculous clusterfuck. In my opinion, it would be good to follow in Mayhem's footsteps here and just not support 4chan's own catalog. That way you could get rid of  all the listed options here, and just basically have the 4chan X Catalog as a normal index mode option like it is right now if you have it enabled. After all, keeping it enabled won't force anyone to use it.

**Comment by Daiz** [comment]
tl;dr/image version
![image](https://cloud.githubusercontent.com/assets/3596343/5934676/2d6aadf2-a6db-11e4-8d88-49a1da3563bb.png)

**Comment by Daiz** [comment]
Alternatively, if you want to keep supporting the native 4chan catalog, the whole thing could still be simplified notably. It could be pretty much the same as suggested above, except you'd keep the "Use 4chan X catalog" in options, and choosing "Catalog" in the index mode would reload the page in native 4chan catalog instead (and then header links there would point to native catalog).

**Comment by ccd0** [comment]
I rarely use anything but paged and catalog mode, so I would be more inclined to have one link to toggle index/catalog and take the catalog option out of the index mode menu.

The "Catalog Links" feature is useful if you're using the native catalog, as otherwise navigating to the index/catalog of a given board will in some cases take two page loads.  If you're using the 4chan X catalog, it's not very useful, and I would recommend turning it off, as is the default.  But I can make it default to the index on the index, the catalog on the catalog, and whichever you used last in threads so it doesn't get in the way if you turn it on.  I think that would work better for people using the native catalog as well.

The simplest option for the current board link would be for it to reload the page, as it would do if 4chan X were disabled.  There's already a button to refresh the index.

**Comment by Daiz** [comment]
>There's already a button to refresh the index.

Which is a tiny button in the middle of other tiny buttons. I found it much more preferable to click on the nice and big board name instead, and seeing as this used to function as an index refresh before, I don't see why it couldn't go back to that. (As a sidenote, I mostly started using the board name link for refreshing in the first place because the refresh button was on the opposite site of the header bar compared to Mayhem's 4chan X, which really fucked with muscle memory of moving my cursor to top left for index refresh.)

**Comment by bakugo** [comment]
>Which is a tiny button in the middle of other tiny buttons.
Completely agree. It's really annoying having to move my mouse over the tiny button, because of this I've started using the keybind but it would be nice to have a bigger button like the [Update] button that shows up on the bottom of threads.

Another thing regarding the catalog: why not just make it another index mode? Right now in the top navigation links there's both a [Return] and [Catalog] link and pressing the return link always switches back to the paged index which is an annoyance sometimes, why not just make it go to the last index mode used or add an option for it? If I remember correctly, Mayhem's fork had an option to add all different index modes to the header (`current-index` would go to the index with the last index mode selected, `current-mode:"paged"` would force paged mode, `current-mode:"catalog"` would force catalog mode, etc. along with a -sort option to change the sorting mode) which was much more convenient.

**Comment by ccd0** [comment]
Reflecting on it some more, while letting the board link reload the whole page would be the simpler option, a soft refresh would definitely be the better option.  I could have it refresh if you're already on the front page, and soft-navigate to the front page without refreshing (unless you have `Refreshed Navigation` turned on) if you aren't.

> Right now in the top navigation links there's both a [Return] and [Catalog] link and pressing the return link always switches back to the paged index which is an annoyance sometimes

How about changing it to [Return] [Index] if you came from the catalog?

And perhaps we could have [Refresh] [Index] in the catalog.

> If I remember correctly, Mayhem's fork had an option to add all different index modes to the header

We do have `current-index` and `current-catalog`, and at some point I plan to add `current-paged` etc. (although you can get that now with `external-text:"Paged","#paged" external-text:"Infinite","#infinite" external-text:"All Pages","#all-pages"`) and `current-bump` etc. (no workaround to get that in the current version yet). That's only there if you're using custom board navigation, though.

**Comment by hotdogman** [comment]
If you do end up completely dropping support for 4chan's native catalog, it would be much appreciated you added #241 to the 4chanX catalog before you dropped it.

**Comment by ccd0** [comment]
The main problem with making the current board link refresh the image is that it's inconsistent with how the index mode-changing links like `current-catalog` work. These shouldn't force an index refresh (although you can get one by turning on `Refreshed Navigation`).

Right now I'm leaning toward

> How about changing it to [Return] [Index] if you came from the catalog?
> And perhaps we could have [Refresh] [Index] in the catalog.

But I do want to get rid of the hard refresh; it should be possible to use the board link to navigate to the front page without a refresh.

Perhaps we could have a `-refresh` modifier for custom navigation?  So you could add links like `current-refresh` and `current-catalog-refresh`?

**Comment by ccd0** [comment]
I added a refresh button (although now I'm thinking it should be unified with the [X minutes ago] text).  I don't think turning the board link into a refresh button would be a good idea because it would make things inconsistent and confusing.
